### PR TITLE
Networking

### LINKED ISSUE #1: Investigate networking

### PR DESCRIPTION
**Pull request by sornas**
There are still some things left to do:

- [x] Error handling -- return `Value::Bool` signaling success or not instead of panicking.
- [x] Handle disconnects
- [x] Send data from server to clients.
- [ ] Simple typechecking of received callable and received parameters. -- I'm thinking that we can do a quick check that the callable matches the parameters received.
- [x] Some detailed comments where needed.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,8 +1,15 @@
-use std::path::Path;
+use std::path::PathBuf;
 
 fn main() {
     let mut args = sylt::Args::parse_args_default_or_exit();
-    args.file = Some(Path::new("game.sy").to_path_buf());
+    args.file = Some(PathBuf::from(
+        std::env::args()
+        .nth(1)
```
Comment (FredTheDino) [inline src/main.rs]: Maybe find the first argument ending in '.sy'? I think I usually write the flags first - and it gets frustrating if it overwrites it - so just swap this out for a `find`?

File: sylt/sylt-common/src/owned_value.rs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    String(String),
+    Function(Vec<OwnedUpValue>, Type, usize),
+    ExternFunction(usize),
+    Unknown,
+    Nil,
+}
+
+impl PartialEq for OwnedValue {
+    fn eq(&self, other: &Self) -> bool {
+        Value::from(self.clone()).eq(&Value::from(other.clone()))
```
Comment (FredTheDino) [inline sylt/sylt-common/src/owned_value.rs:32]: ;)

File: sylt/sylt-std/src/network.rs [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+    ))
+}
+
+/// Parse args given to an external function as rpc arguments, i.e. one callable followed by 0..n arguments.
+fn get_rpc_args(ctx: RuntimeContext<'_>, func_name: &str) -> Result<(OwnedValue, OwnedValue), RuntimeError> {
+    let values = ctx.machine.stack_from_base(ctx.stack_base);
+    let owned_values: Vec<OwnedValue> = values.iter().map(|v| v.into()).collect();
+
+    if owned_values.len() != 0 {
+        let (callable, args) = owned_values.split_at(1);
```
Comment (FredTheDino) [inline sylt/sylt-std/src/network.rs:180]: Nice

File: sylt/sylt-std/src/network.rs [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+
+//TODO(gu): This doc is wrong since this takes variadic arguments.
+#[sylt_macro::sylt_doc(n_rpc_server, "Performs an RPC on the connected server, returning success status.", [One(Value(callable)), One(Value(args))] Type::Bool)]
+#[sylt_macro::sylt_link(n_rpc_server, "sylt_std::network")]
+pub fn n_rpc_server(ctx: RuntimeContext<'_>) -> Result<Value, RuntimeError> {
+    if ctx.typecheck {
+        return Ok(Value::Bool(true));
+    }
+
+    // Serialice the RPC.
```
Comment (FredTheDino) [inline sylt/sylt-std/src/network.rs]:
```suggestion
    // Serialize the RPC.
```

**Review by FredTheDino — APPROVED** [review]

File: sylt/sylt-common/src/owned_value.rs (hunk 2 above)
Comment (sornas) [reply sylt/sylt-common/src/owned_value.rs:32]: Yeah. It Just Works :tm:.

**Pull request merged**